### PR TITLE
See Certificates links for non FA courses

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Related to #3233

#### What's this PR do?
Show certificate link for edx certificates for non FA courses.

#### How should this be manually tested?
Can test this only if there is a verified certificate on edx.
Can create a `CachedCertificate`:
```
cert_json = {
            "username": username,
            "course_id": course_key,
            "certificate_type": "verified",
            "is_passing": True,
            "status": "downloadable",
            "download_url": some_url,
            "grade": "0.98"
        }
CachedCertificate.deserialize_edx_data([cert_json])
```

### CODE REVIEW

File: dashboard/api.py [hunk 1]
```diff
@@ -4,8 +4,10 @@
 import datetime
 import logging
 
+from urllib.parse import urljoin
```
Comment (giocalitri) [inline dashboard/api.py]: nit: this belongs to the standard library import group

File: dashboard/api.py [hunk 2]
```diff
@@ -4,8 +4,10 @@
...
 from django.core.exceptions import ImproperlyConfigured
 from django.db import transaction
+from django.conf import settings
```
Comment (giocalitri) [inline dashboard/api.py]: nit: please sort the imports

File: dashboard/api.py [hunk 3]
```diff
@@ -479,10 +481,16 @@ def get_certificate_url(mmtrack, course):
...
     url = ""
+    final_grades = mmtrack.get_passing_final_grades_for_course(course)
     if final_grades.exists():
         best_grade = final_grades.first()
-        if best_grade.has_certificate:
-            url = "/certificate/{}".format(final_grades.first().certificate.hash)
+        course_key = best_grade.course_run.edx_course_key
+        if mmtrack.financial_aid_available:
+            if best_grade.has_certificate:
+                url = "/certificate/{}".format(final_grades.first().certificate.hash)
```
Comment (giocalitri) [inline dashboard/api.py]: Cannot create this url using the `reverse`?

File: dashboard/api.py [hunk 4]
```diff
@@ -4,11 +4,12 @@
...
 import logging
 
-from urllib.parse import urljoin
+from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 from django.db import transaction
-from django.conf import settings
+from django.urls import reverse
 import pytz
+from urllib.parse import urljoin
```
Comment (giocalitri) [inline dashboard/api.py]: nit: the `import` and the `from <> import` need to be grouped together

**Comment by annagav** [comment]
@giocalitri, addressed all comments

**Comment by annagav** [comment]
@giocalitri, I will fix the test if you approve the change to refresh certificates.

**Comment by pdpinch** [comment]
Why hasn't this merged yet? I'm actually not in a rush, but I'd like to know what's holding it up.

File: dashboard/api.py [hunk 5]
```diff
@@ -133,8 +136,10 @@ def get_user_program_info(user, edx_client):
     # NOTE: this part can be moved to an asynchronous task
     if edx_client is not None:
         try:
-            for cache_type in CachedEdxDataApi.SUPPORTED_CACHES:
+            for cache_type in [CachedEdxDataApi.ENROLLMENT, CachedEdxDataApi.CURRENT_GRADE]:
```
Comment (giocalitri) [inline dashboard/api.py]: Why this change? Cannot this might effect the part of the dashboard?

File: dashboard/api.py [hunk 6]
```diff
@@ -479,10 +482,16 @@ def get_certificate_url(mmtrack, course):
...
     url = ""
+    final_grades = mmtrack.get_passing_final_grades_for_course(course)
     if final_grades.exists():
         best_grade = final_grades.first()
-        if best_grade.has_certificate:
-            url = "/certificate/{}".format(final_grades.first().certificate.hash)
+        course_key = best_grade.course_run.edx_course_key
+        if mmtrack.financial_aid_available:
+            if best_grade.has_certificate:
+                url = reverse('certificate', args=[final_grades.first().certificate.hash])
```
Comment (giocalitri) [inline dashboard/api.py]: isn't `final_grades.first()` equal to `best_grade`?

Comment (annagav) [reply dashboard/api.py]: yes, it is

**Comment by annagav** [comment]
addressed all comments! @giocalitri
